manager/allocator/networkallocator/networkallocator.go

```diff
@@ -186,12 +187,28 @@ func (na *NetworkAllocator) ServiceAllocate(s *api.Service) (err error) {
 		return
 	}
 
-	// First allocate VIPs for all the pre-populated endpoint attachments
-	for _, eAttach := range s.Endpoint.VirtualIPs {
-		if err = na.allocateVIP(eAttach); err != nil {
-			return
+	// Allocate VIPs for all the pre-populated endpoint attachments
+	eVIPs := s.Endpoint.VirtualIPs[:0]
+	var match bool
+	for idx, eAttach := range s.Endpoint.VirtualIPs {
+		match = false
+		for _, nAttach := range s.Spec.Task.Networks {
```

Comment:
Instead of accessing `s.Spec.Task.Networks` directly, move the following code from below above this, and use `specNetworks`:

```go
        // Always prefer NetworkAttachmentConfig in the TaskSpec
        specNetworks := s.Spec.Task.Networks
        if len(specNetworks) == 0 && s != nil && len(s.Spec.Networks) != 0 {
                specNetworks = s.Spec.Networks
        }
```